src/main/java/world/bentobox/parkour/listeners/CourseRunnerListener.java

```diff
@@ -113,27 +114,32 @@ public void onVisitorFall(EntityDamageEvent e) {
         player.playEffect(EntityEffect.ENTITY_POOF);
         player.setVelocity(new Vector(0, 0, 0));
         player.setFallDistance(0);
-        player.teleport(parkourRunManager.checkpoints().get(player.getUniqueId()));
+        Location checkpointLocation = parkourRunManager.checkpoints().get(player.getUniqueId());
+        checkpointLocation = checkpointLocation.clone().add(0.5, 0, 0.5);
+        parkourRunManager.currentlyTeleporting().add(player.getUniqueId());
+        player.teleport(checkpointLocation);
```

Comment:
I think we may need to use teleportAsync(@NonNull Entity entity, @NonNull Location location, TeleportCause cause) instead of teleport. It's probably not a big deal as the checkpoint chunk is likely to be loaded, but maybe on a big map, or maybe if the player goes between dimensions, it could help.